src/Directory.BeforeCommon.targets

```diff
@@ -108,10 +108,10 @@
     <DefineConstants Condition="'$(MonoBuild)'=='true'">$(DefineConstants);MONO</DefineConstants>
 
     <!-- Useful for tests that need OS information as a constant -->
-    <DefineConstants Condition="'$(OsEnvironment)'=='windows'">$(DefineConstants);TEST_ISWINDOWS</DefineConstants>
+    <DefineConstants Condition="$([MSBuild]::IsOSPlatform('windows'))">$(DefineConstants);TEST_ISWINDOWS</DefineConstants>
   </PropertyGroup>
 
-  <PropertyGroup Condition="$(TargetFramework.StartsWith('netstandard')) or $(TargetFramework.StartsWith('netcore'))">
+  <PropertyGroup Condition="$(TargetFramework.StartsWith('netstandard')) or $(TargetFramework.StartsWith('netcore')) or $(TargetFramework.StartsWith('net5'))">
```

Comment:
Replace with is-compatible-with-netcoreapp1.0 property function?

Reply:
Are all netcoreapp1.0+ target frameworks backwards compatible that far?

Reply:
theoretically yes

Reply:
Well this is strange. Changing this to: `$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'netcoreapp1.0'))` results in a compilation error:
```
C:\src\git\msbuild\src\Tasks\AssemblyDependency\AssemblyInformation.cs(34,26): error CS0246: The type or namespace name
 'IMetaDataDispenser' could not be found (are you missing a using directive or an assembly reference?) [C:\src\git\msbu
ild\src\Tasks\Microsoft.Build.Tasks.csproj]
C:\src\git\msbuild\src\Tasks\AssemblyDependency\AssemblyInformation.cs(35,26): error CS0246: The type or namespace name
 'IMetaDataAssemblyImport' could not be found (are you missing a using directive or an assembly reference?)
and others
```
The failing project is Microsoft.Build.Tasks.csproj targeting netstandard2.0. This condition must have been false for netstandard2.0 previously? 

Here's a sample of the failing code:
```c#
#if !FEATURE_ASSEMBLYLOADCONTEXT
            if (NativeMethodsShared.IsWindows)
            {
                // Create the metadata dispenser and open scope on the source file.
                _metadataDispenser = (IMetaDataDispenser)new CorMetaDataDispenser();
                _assemblyImport = (IMetaDataAssemblyImport)_metadataDispenser.OpenScope(sourceFile, 0, ref s_importerGuid);
            }
            else
            {
                _assembly = Assembly.ReflectionOnlyLoadFrom(sourceFile);
            }
#endif

```

My first assumption would be that the netstandard2.0 project didn't have FEATURE_ASSEMBLYLOADCONTEXT defined, and now it is and doesn't see those types.
- This is even more confusing considering the previous check was checking if the TargetFramework started with `netstandard`, so it would've fallen into this case anyway?

So it seems like there are two paths here:
- Get the type to show up in the netstandard2.0 target (which doesn't seem right to me considering it apparently wasn't visible before),
- Compare the two binlogs and see what's different between the current condition and the previous.
- - Ex: See if the previous condition had FEATURE_ASSEMBLYLOADCONTEXT defined.

Reply:
I think I see the problem. `netcoreapp1.0` is not compatible with `netstandard2.0`, so I think I need to change the condition to "is target framework compatible with netcoreapp1.0 OR does the target framework start with netstandard"

Reply:
I figured that by seeing what constants were defined in a build with the old condition vs the new condition.
```
Working (Old condition)
----
TRACE
MICROSOFT_BUILD_TASKS
STANDALONEBUILD
FEATURE_DEBUG_LAUNCH
TEST_ISWINDOWS
RUNTIME_TYPE_NETCORE
FEATURE_ASSEMBLYLOADCONTEXT
FEATURE_PROCESSSTARTINFO_ENVIRONMENT
FEATURE_RUNTIMEINFORMATION
USE_MSBUILD_DLL_EXTN
WORKAROUND_COREFX_19110
DEBUG
NETSTANDARD
NETSTANDARD2_0

Not Working (new condition)
----
TRACE
MICROSOFT_BUILD_TASKS
STANDALONEBUILD
FEATURE_DEBUG_LAUNCH
TEST_ISWINDOWS
DEBUG
NETSTANDARD
NETSTANDARD2_0

```

Reply:
And that fixed it 👍